com.arc.embeddedcdt/src/com/arc/embeddedcdt/common/DebuggerPluginVersionsChecker.java

```diff
@@ -0,0 +1,61 @@
+package com.arc.embeddedcdt.common;
+
+import java.util.HashSet;
+
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.swt.widgets.Display;
+
+import com.arc.embeddedcdt.LaunchConfigurationConstants;
+import com.arc.embeddedcdt.dsf.utils.Configuration;
+
+public class DebuggerPluginVersionsChecker {
+
+    private HashSet<String> seenLaunchConfigurationNames = new HashSet<>();
+
+    private static DebuggerPluginVersionsChecker INSTANCE = null;
+
+    private ILaunchConfiguration launchConfiguration;
+
+    private DebuggerPluginVersionsChecker() {
+    }
+
+    public static DebuggerPluginVersionsChecker getDebuggerPluginVersionsChecker() {
+        if (INSTANCE == null) {
+            INSTANCE = new DebuggerPluginVersionsChecker();
+        }
+        return INSTANCE;
+    }
+
+    public void checkPluginVersion(ILaunchConfiguration launchConfiguration) {
+        String launchConfigurationName = Configuration.getProgramName(launchConfiguration);
+        if (!seenLaunchConfigurationNames.contains(launchConfigurationName)) {
+            seenLaunchConfigurationNames.add(launchConfigurationName);
+            final int currentLaunchConfigurationDebuggerPluginVersion = Configuration
+                    .getDebuggerPluginVersion(launchConfiguration);
+            if (currentLaunchConfigurationDebuggerPluginVersion != LaunchConfigurationConstants.UNREAL_DEBUGGER_PLUGIN_VERSION_NUMBER
```

Comment:
You need to explain in the comment, which values are valid, which are not. From the code it looks like that if file doesn't contain version number it will be accepted - this is not good, because new plugin will accept any of the existing launch files, which we actually aim to prevent with this patch.